### DISCUSSION TITLE
I wonder is there a tool could record the execution time of a wasm func

### DISCUSSION THREAD

**Discussion by giteewif** [bytecodealliance/wasm-tools#1885] [Q&A]
Record execution time of a wasm func, or insert the bench record code before or after the func call in wasm,
 there is a project like this wasabi which depends on wasmparser and wasm-bindgen

**Comment by alexcrichton** [comment]
Currently this repository doesn't have such a tool built-in already, but as noted it should be possible to build one using the crates in this repository.

**Comment by giteewif** [comment]
The perfmap or jitmap profiling of  wasmtime based on perf，could do this, it get the % of the execution time. Insert the bench code seems to cost much of time no so practical.
